subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/model/IdeRepoFileDependency.java

```diff
@@ -33,10 +36,20 @@ public File getFile() {
     }
 
     public ModuleVersionIdentifier getId() {
+        if (componentIdentifier == null) {
+            return null;
+        }
+        if (id == null) {
+            id = new DefaultModuleVersionIdentifier(componentIdentifier.getGroup(), componentIdentifier.getModule(), componentIdentifier.getVersion());
+        }
```

Comment:
Could use `DefaultModuleVersionIdentifier.newId(ModuleComponentIdentifier)`. And perhaps we should make that null-safe.